src/main/java/uk/nhs/prm/repo/ehrtransferservice/config/SnsClientSpringConfiguration.java

```diff
@@ -2,25 +2,32 @@
 
 import com.amazonaws.services.sns.AmazonSNS;
 import com.amazonaws.services.sns.AmazonSNSClientBuilder;
+import lombok.RequiredArgsConstructor;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
 import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.services.sns.SnsClient;
 
 @Configuration
-
+@RequiredArgsConstructor
 public class SnsClientSpringConfiguration {
     @Value("${aws.region}")
     private String awsRegion;
 
+    private AwsCredentialsProvider awsCredentialsProvider;
+
     @Bean
     public AmazonSNS snsClient() {
         return AmazonSNSClientBuilder.standard().withRegion(awsRegion).build();
     }
 
     @Bean
     public SnsClient snsClientV2() {
-        return SnsClient.builder().region(Region.of(awsRegion)).build();
+        return SnsClient.builder()
+                .region(Region.of(awsRegion))
+                .credentialsProvider(awsCredentialsProvider)
+                .build();
```

Comment:
My only concern, could this be problematic? https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/creating-clients.html#:~:text=Explicitly%20shut%20down%20clients%20when%20they%20are%20no%20longer%20needed%20to%20avoid%20resource%20leaks.

I don't believe this _will_ be an issue, just wanted to raise this. Maybe we can spike this
at some point in the future?